src/core/dfn.js

```diff
@@ -26,9 +25,36 @@ export function run() {
     }
 
     // Only add `lt`s that are different from the text content
-    if (titles.length === 1 && titles[0] === norm(dfn.textContent)) {
+    if (titles.length === 1 && firstTitle === norm(dfn.textContent)) {
       return;
     }
     dfn.dataset.lt = titles.join("|");
   });
 }
+/**
+ *
+ * @param {string} title
+ * @param {HTMLElement} dfn
+ */
+function associateInternalSlotWithIDL(title, dfn) {
+  if (!/^\[\[\w+\]\]$/.test(title)) return;
+  if (!dfn.dataset.idl) {
+    dfn.dataset.idl = "";
+  }
+
+  // Automatically use the closest data-dfn-for as the parent.
+  /** @type HTMLElement */
+  const parent = dfn.closest("[data-dfn-for]");
+  if (parent && dfn !== parent && parent.dataset.dfnFor) {
+    dfn.dataset.dfnFor = parent.dataset.dfnFor;
+  }
+
+  // Assure that it's data-dfn-for= something.
+  if (!dfn.dataset.dfnFor) {
+    const msg = `Internal slot "${title}" must be associated with a WebIDL interface.`;
+    const hint =
+      "Use a `data-dfn-for` attribute to associate this dfn with a WebIDL interface.";
+    showError(msg, name, { hint, elements: [dfn] });
+  }
+  dfn.dataset.dfnType = "attribute";
```

Comment:
This is something that will be fixed in follow-up, right? i.e., `method` or a new `internalSlot` type?

Reply:
I don't think so.. See, for example:
https://respec.org/xref/?term=%5B%5BCollectFromCredentialStore%5D%5D%28origin%2C+options%2C+sameOriginWithAncestors%29

Reply:
Will the existing tools throw if we introduce a new `dfnType` here?

Reply:
re new type, discussion started in https://github.com/speced/spec-dfn-contract/issues/1